modules/httpcache/httpcache.go

```diff
@@ -49,6 +49,7 @@ func HandleTimeCache(req *http.Request, w http.ResponseWriter, fi os.FileInfo) (
 func HandleEtagCache(req *http.Request, w http.ResponseWriter, fi os.FileInfo) (handled bool) {
 	etag := generateETag(fi)
 	if req.Header.Get("If-None-Match") == etag {
+		w.Header().Set("ETag", etag)
```

Comment:
`Etag` is already set below for non-matching requests. What is the benefit of echoing the same Etag back to the client that was already sent?

Reply:
https://tools.ietf.org/html/rfc7232#section-4.1
```
The server generating a 304 response MUST generate any of the following header fields that would have been sent in a 200 (OK) response to the same request: Cache-Control, Content-Location, Date, ETag, Expires, and Vary.
```

Reply:
Okay in that case, you can move the `Header().Set` before the `if`. Could you please do the same for `HandleTimeCache` above with `Cache-Control` to fix it there too?

Reply:
Fixed with dd68722